tests/include/test/macros.h

```diff
@@ -123,36 +123,36 @@
  *                  This expression may be evaluated multiple times.
  *
  */
-#define ASSERT_ALLOC(pointer, length)                           \
-    do                                                            \
-    {                                                             \
-        TEST_ASSERT((pointer) == NULL);                       \
-        if ((length) != 0)                                     \
-        {                                                         \
-            (pointer) = mbedtls_calloc(sizeof(*(pointer)), \
+#define TEST_CALLOC_OR_FAIL(pointer, length)                \
```

Comment:
I'm still not convinced that it's better to have `CALLOC` in the name than `ALLOC`. This macro behaves similarly to `calloc`, but it doesn't take input in the same way: its first argument is the pointer.

Reply:
Agreed, but we have often had confusion where people think the length (now `item_count` - good suggestion) was bytes, which leads to ineffective sanitiser runs.

Apart from how the item size is determined, this behaves more like `calloc()` than `malloc()`, with the multiplying to get actual number of bytes allocated, and zeroing.